internal/hashutil/hashutil.go

```diff
@@ -27,3 +29,167 @@ func Hash(v any) (uint64, error) {
 	}
 	return hashstructure.Hash(v, hashstructure.FormatV2, opts)
 }
+
+// MapHash efficiently computes a non-cryptographic hash of a map of strings.
+func MapHash(iv uint64, m map[string]string) uint64 {
+	accum := iv
+	var lenBuf [4]byte
+	binary.BigEndian.PutUint32(lenBuf[:], uint32(len(m)))
+	accum ^= xxhash.Sum64(lenBuf[:])
+	var kvBuf [16]byte
+	for k, v := range m {
```

Comment:
Is this using the same trick as hashstructure? XORing all of them together so iteration order doesn't matter?

Reply:
Yes, it is the same except this allows for domain separation using a custom initialization value for the hash. this is done because we remove the map fields before performing proto serialization (since they get copied/sorted internally when using deterministic mode) then add the hashes of those maps at the end. this would mess up the field ordering so the hash of the field name is used as the starting value to distinguish between multiple map fields in a single message.